src/app/features/user/delete-account/delete-account.component.ts

```diff
@@ -0,0 +1,81 @@
+import { Component, OnDestroy, OnInit } from '@angular/core'
+import { FormBuilder, FormGroup } from '@angular/forms'
+import { Router } from '@angular/router'
+import { User } from '@core/models'
+import {
+  AlertService,
+  AuthenticationService,
+  UserService,
+} from '@core/services'
+import { Subscription } from 'rxjs'
+import { first } from 'rxjs/operators'
+import { FieldSpecs } from '@app/shared/validation/field-spec'
+
+@Component({
+  selector: 'app-delete-account',
+  templateUrl: './delete-account.component.html',
+  styleUrls: ['./delete-account.component.scss'],
+})
+export class DeleteAccountComponent implements OnInit, OnDestroy {
+  currentUser: User
+  currentUserSubscription: Subscription
+  loading = false
+  submitted = false
+  deleted = false
+  passwordForm: FormGroup
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private authenticateService: AuthenticationService,
+    private userService: UserService,
+    private alertService: AlertService,
+    private router: Router
+  ) {}
+
+  ngOnInit() {
+    this.currentUserSubscription = this.authenticateService.currentUser.subscribe(
+      user => {
+        this.currentUser = user
+      }
+    )
+
+    this.passwordForm = this.formBuilder.group({
+      password: ['', FieldSpecs.passwordValidator],
+    })
+  }
+
+  ngOnDestroy() {
+    this.currentUserSubscription.unsubscribe()
```

Comment:
need to check this.currentUserSubscription is null or undefined before doing unsubscribe()